packages/shared_preferences/shared_preferences_android/lib/src/shared_preferences_android.dart

```diff
@@ -39,20 +41,27 @@ class SharedPreferencesAndroid extends SharedPreferencesStorePlatform {
   Future<bool> setValue(String valueType, String key, Object value) async {
     switch (valueType) {
       case 'String':
-        return _api.setString(key, value as String);
+        value as String;
+        if (value.startsWith(listPrefix) ||
```

Comment:
Will this list grow? if so consider making a set of prefixes to check.

Reply:
We don't generally add types to `shared_preferences` since anything that all platforms don't support ends up with unpleasant complexity like the `List<String>` mess we are currently dealing with.